modules/gridmap/grid_map.cpp

```diff
@@ -409,6 +409,26 @@ int GridMap::get_cell_item_orientation(const Vector3i &p_position) const {
 	return cell_map[key].rot;
 }
 
+Basis GridMap::get_cell_item_basis(const Vector3i &p_position) const {
+	int orientation = get_cell_item_orientation(p_position);
+
+	if (orientation == -1) {
+		return Basis();
+	}
+
+	return get_basis_with_orthogonal_index(orientation);
+}
+
+Basis GridMap::get_basis_with_orthogonal_index(int p_index) const {
+	Basis basis;
+	basis.set_orthogonal_index(p_index);
+	return basis;
+}
```

Comment:
I meant that the whole `Basis::set_orthogonal_index` should be moved to GridMap, as it's not needed to be in `Basis` itself.